oxide-openapi-gen-ts/src/client/static/http-client.test.ts

```diff
@@ -17,6 +17,41 @@ const headers = { "Content-Type": "application/json" };
 const json = (body: any, status = 200) =>
   new Response(JSON.stringify(body), { status, headers });
 
+describe("fetchWithRetry", () => {
+  it("retries request when handler returns true", async () => {
+    const retryLimit = 1
+    let retries = 0
+    const retryHandler = () => {
+      if (retries >= retryLimit) {
+        return false
+      } else {
+        retries += 1
+        return true
+      }
+    }
```

Comment:
Will review in more detail, but my immediate reaction is that as implemented, the retry logic is still entirely outside the client and `fetchWithRetry` doesn't add that much. I'll try and put together a version of this that doesn't require `fetchWithRetry` and we can compare.

Here's an example in the console that uses the [`p-retry`](https://github.com/sindresorhus/p-retry) library. It's not exactly analogous, but the `mutateAsync` call is very close to calling the client method directly.

https://github.com/oxidecomputer/console/blob/1625d02a540035e6b10e6222e7ef709691dda7f1/app/forms/image-upload.tsx#L424-L429